packages/liquid-html-parser/src/stage-1-cst.spec.ts

```diff
@@ -1108,6 +1108,83 @@ describe('Unit: Stage 1 (CST)', () => {
           expectPath(cst, '0.children.2.type').to.equal('TextNode');
           expectPath(cst, '0.children.2.value').to.equal('@unsupported');
         });
+
+        it('should parse a basic example tag', () => {
+          const testStr = `{% doc -%} @example {%- enddoc %}`;
+          cst = toCST(testStr);
+          expectPath(cst, '0.type').to.equal('LiquidRawTag');
+          expectPath(cst, '0.name').to.equal('doc');
+          expectPath(cst, '0.children.0.type').to.equal('LiquidDocExampleNode');
+          expectPath(cst, '0.children.0.exampleContent.value').to.equal('');
+        });
+
+        it('should parse example tag with content that has leading whitespace', () => {
+          const testStr = `{% doc %} @example         hello there       {%- enddoc %}`;
+          cst = toCST(testStr);
+          expectPath(cst, '0.type').to.equal('LiquidRawTag');
+          expectPath(cst, '0.name').to.equal('doc');
+          expectPath(cst, '0.children.0.type').to.equal('LiquidDocExampleNode');
+          expectPath(cst, '0.children.0.name').to.equal('example');
+          expectPath(cst, '0.children.0.exampleContent.value').to.equal('hello there');
```

Comment:
let's add some tests for locStart / locEnd and document how we want that to behave, especially considering whitespace etc

Can also add some assertions in stage-2 for position